cli/pkg/workspace/plugin/kubeconfig.go

```diff
@@ -132,187 +134,212 @@ func (o *UseWorkspaceOptions) Run(ctx context.Context) error {
+	}
 
-	case "..":
-		config, err := o.ClientConfig.ClientConfig()
-		if err != nil {
-			return err
-		}
-		u, currentClusterName, err := pluginhelpers.ParseClusterURL(config.Host)
-		if err != nil {
-			return fmt.Errorf("current URL %q does not point to a workspace", config.Host)
-		}
-		parentClusterName, hasParent := currentClusterName.Parent()
-		if !hasParent {
-			if currentClusterName == core.RootCluster.Path() {
-				return fmt.Errorf("current workspace is %q", currentClusterName)
-			}
-			return fmt.Errorf("current workspace %q has no parent", currentClusterName)
-		}
-		u.Path = path.Join(u.Path, parentClusterName.RequestPath())
-		newServerHost = u.String()
+	// validate early for valid name
+	err = o.validate(ctx)
+	if err != nil {
+		return err
+	}
 
-	case "":
+	switch {
+	// o.Name empty and home workspace should be always one after another for fallthrough to
+	// print error and default to home.
+	case o.Name == "":
 		defer func() {
 			if err == nil {
 				_, err = fmt.Fprintf(o.Out, "Note: 'kubectl ws' now matches 'cd' semantics: go to home workspace. 'kubectl ws -' to go back. 'kubectl ws .' to print current workspace.\n")
 			}
 		}()
 		fallthrough
 
-	case "~", home:
-		homeWorkspace, err := o.kcpClusterClient.Cluster(core.RootCluster.Path()).TenancyV1alpha1().Workspaces().Get(ctx, "~", metav1.GetOptions{})
+	case o.Name == "~" || o.Name == home || strings.HasPrefix(o.Name, "~:") || strings.HasPrefix(o.Name, home):
+		newServerHost, err := o.moveHome(ctx, home)
 		if err != nil {
 			return err
 		}
-		config, err := o.ClientConfig.ClientConfig()
+		return o.commitConfig(ctx, currentContext, newServerHost, nil)
+
+		// move to 'my' root
+	case o.Name == ":":
+		newServerHost, err := o.moveCurrentRoot(ctx)
 		if err != nil {
 			return err
 		}
-		u, _, err := pluginhelpers.ParseClusterURL(config.Host)
+		return o.commitConfig(ctx, currentContext, newServerHost, nil)
+
+	case o.Name == ".":
+		cfg, err := o.ClientConfig.ClientConfig()
 		if err != nil {
-			return fmt.Errorf("current URL %q does not point to a workspace", config.Host)
+			return err
 		}
-		uh, err := url.Parse(homeWorkspace.Spec.URL)
+		return printCurrentWorkspace(o.Out, cfg.Host, shortWorkspaceOutput(o.ShortWorkspaceOutput), nil)
+
+	case o.Name == "-":
+		newServerHost, err := o.movePrevious(ctx, currentContext)
 		if err != nil {
-			return fmt.Errorf("invalid home workspace URL %q: %w", homeWorkspace.Spec.URL, err)
+			return err
 		}
+		return printCurrentWorkspace(o.Out, newServerHost, shortWorkspaceOutput(o.ShortWorkspaceOutput), nil)
 
-		// We keep the old host, and append homeworkspace url. This allows users to have front-proxy
-		// host to be different from the workspace host.
-		newServerHost = u.Scheme + "://" + path.Join(u.Host, uh.Path)
-
-	case ".":
-		cfg, err := o.ClientConfig.ClientConfig()
+	case strings.Contains(o.Name, ".."):
```

Comment:
this one looks fishy. Could see `= ".."`, but anything beyond that needs some more logic, e.g. for `./../foo/../bar`.

Reply:
yes, this is not supported yet. Code code quite convoluted and wanna move it to separate pr. So this is not yet supported